### PR TITLE
Reduce memory from TabularPandas

### PR DESCRIPTION
**Pull request by muellerzr**
As discussed [here](https://github.com/fastai/fastai2/issues/148), this was my first thought on it. Let me know if you were thinking of something a little bit different, or if I should change anything at all. (I also added a check to see if we already set the pandas chained option)

### CODE REVIEW

**Comment by muellerzr** [comment]
Don't merge this quite yet please, unexpected bug :)

**Comment by muellerzr** [comment]
Fixed the bug, testing on Rossmann footprint was reduced from 2.6gb to 2.15gb (Matching the experiments)

**Comment by sgugger** [comment]
This looks good but you have quite a few weird exports. Do you have the latest `nbdev`?

**Comment by muellerzr** [comment]
Good catch, I do not! Let me try again

**Comment by muellerzr** [comment]
Looks as though that fixed it all (except metrics.py for some reason)

**Comment by sgugger** [comment]
Better, thanks! Does this close #148?

**Comment by muellerzr** [comment]
Yes it does!

**Comment by sgugger** [comment]
The `reduce_cats` make the tests later in the book fail (I commented it out for now). The problem is that the categories should be decided on the training set (if there is one) and then used on the validation set (which is the whole point of `Categorify`)

**Comment by muellerzr** [comment]
Got it, yea there was a bug pointed out earlier on the forums about this. I'll make the adjustment I think is needed and push a PR shortly
